Add ability to launch another humility command along with qemu

beezow opened the pull request:
Adds ability to launch qemu, then another command a certain time after. ex: `humility qemu -c tasks --delay 500` will run `humility tasks` 500ms after qemu launches, then exit.  Additionally `humility qemu -c tasks --delay 500 --silent` would hide the `qemu` stdout and only show `tasks` stdout.  
The primary motivation for this is being able to easily run commands targeting qemu cores within the humility test suite.  Additionally, when testing commands, I often find myself following the same pattern.

~~This is marked to merge into `dev/drew/gdb_ports` to represent the dependency on https://github.com/rivosinc/humility/pull/34.~~  Without being able to specify the port, there would be no guarantee we are connecting to the right qemu instance.

richard-rivos commented on cmd/qemu/src/lib.rs:
```diff
@@ -117,12 +145,52 @@ fn qemu(context: &mut humility::ExecutionContext) -> Result<()> {
+        } else if let Some(command) = subargs.command {
+            // we unfornunatly have to contruct a new command from scratch, calling back into the
+            // base humility command parsers would create a circular dependency
+            let mut cmd = Command::new("humility");
```
This assumes 'humility' is available on 'PATH'.  This may not necessarily be the case (or the version on humility currently executing may not be the same as the one found on the path. Ideally, we should try and make sure we invoke the same humility as is currently running.

FawazTirmizi replied:
Looks like `std::env::current_exe()` can provide the path of the current-running executable: https://doc.rust-lang.org/std/env/fn.current_exe.html.

beezow replied:
Fixed.

beezow commented:
I changed the default delay, ~~added a print statement to reflect the exact command that will be run~~, and changed it to ensure the same executable will be used.

beezow commented:
Just rebased onto main.
